x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java

```diff
@@ -192,56 +191,44 @@ private void createFollowerIndex(
             threadPool.getThreadContext().getHeaders(),
             clusterService.state()
         );
-        threadPool.executor(ThreadPool.Names.SNAPSHOT).execute(new AbstractRunnable() {
```

Comment:
Not sure I follow why this is important to this PR?

Reply:
I kinda liked cleaning this up here since it's part of the necessary follow-up fixes for the async behaviour to work neatly in a sense, but I can pull it out to a separate PR if you want?

Reply:
https://github.com/elastic/elasticsearch/pull/91405